DynamicArray/DynamicIntArray.cpp

```diff
@@ -0,0 +1,115 @@
+#include "DynamicIntArray.h"
+
+//Implementations go here
+
+DynamicIntArray::DynamicIntArray()
+{
+	size = 0;
+	array = nullptr;
+}
+
+DynamicIntArray::DynamicIntArray(std::size_t size)
+{
+	this->size = size;
+	array = new int[size];
+}
+
+DynamicIntArray::DynamicIntArray(const DynamicIntArray& other)
+{
+	size = other.size;
+	array = new int[size];
+	for (std::size_t i = 0; i < size; i++)
+	{
+		array[i] = other.array[i];
+	}
+}
+
+DynamicIntArray::~DynamicIntArray()
+{
+	delete[] array;
+}
+
+DynamicIntArray& DynamicIntArray::operator=(const DynamicIntArray& other)
+{
+	if(this == &other) {
+		return *this;
+	}
+
+	delete[] array; 
+
+	size = other.size;
+	array = new int[other.size];
+	for (size_t i = 0; i < other.size; i++)
+	{
+		array[i] = other.array[i]; 
+	}
+
+	return *this; 
+
+}
+
+int& DynamicIntArray::operator[](std::size_t index)
+{
+	return array[index];
+}
+
+void DynamicIntArray::setSize(std::size_t newSize)
+{
+	delete[] array;
+	size = newSize;
+	array = new int[size];
+}
```

Comment:
You shouldn't just unconditionally delete the array clearing any content there was.
E.g., f the size is bigger than you can safely safe ALL old content as it was

Reply:
Corrected